### ISSUE TITLE
Decide how to handle assertions, leaks, infra issues

### ISSUE THREAD

**Issue by marco-c** [mozilla/mozci#686]
I think there are other things to consider:
 * assertion
 * leak
 * infra

on timeout, do we care about test timeout or tooling/harness/browser timeout?

_Originally posted by @jmaher in https://github.com/mozilla/mozci/issues/684#issuecomment-1065138011_

**Comment by marco-c** [comment]
Should we identify them differently than crashes, timeouts, test assertion failures?

I guess assertion could be considered in the same way as a crash, and maybe leaks too?

infra I'm not sure we are able to precisely detect, as infra issues might mean the artifacts we usually parse are somehow broken.

**Comment by marco-c** [comment]
The problem I see with these issues that are not test assertion failures is that they might happen across many tests.
So, we can't only track them as failures associated to each test, otherwise the signal about their frequency would be lost.

If we have a crash happening across 100 tests and we increase the intermittency frequency for all 100 tests, we end up with 100 tests with 1 failure, when in reality we have one crash with 100 instances.

**Comment by jmaher** [comment]
yes- crashes should be separate, I would argue that assertions should be as well.  timeouts can be test independent, but typically if a test name is associated with it, then it is the failure.

One thing to ensure we be careful with is first failure vs all failures.  Often the first failure is the root cause and other failures are side effects of the first failure.  This is very common in browser-chrome and devtools, but can be seen in other tests (even xpcshell).  But to add to that, if we want the crash, then possibly the crash is not the first failure.

On the topic of crashes...if the harness times out, then we will terminate the browser, on some platforms (I think windows) this results in a crash of the browser.  This isn't a crash as much as it is us trying to cleanup and then run the next set of tests.
